src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
+
+  public getUserData(userId: string): string {
+    const query = `SELECT * FROM users WHERE id = '${userId}'`;
+    return this.executeQuery(query);
+  }
```

Comment:
PullProofAI

Avoid directly interpolating user input into SQL queries to prevent SQL injection attacks. Instead, use parameterized queries or an ORM to handle database interactions securely.

#### `findDuplicates` Method


<!-- This is an auto-generated comment by PullProofAI -->